README.md

```diff
@@ -1512,68 +1512,126 @@ const car = new Car()
+- ["3 Different Kinds of Prototypal Inheritance: ES6+ Edition"](https://medium.com/javascript-scene/3-different-kinds-of-prototypal-inheritance-es6-edition-32d777fa16c9)
+- ["What's the Difference Between Class and Prototypal Inheritance?"](https://medium.com/javascript-scene/master-the-javascript-interview-what-s-the-difference-between-class-prototypal-inheritance-e4cd0a7562e9)
+- [You Don't Know JS: Simpler design](https://github.com/getify/You-Dont-Know-JS/blob/master/this%20%26%20object%20prototypes/ch6.md#simpler-design)
+
+After you read mentioned articles you know that JavaScript uses **prototypal inheritance** and you as a developer should favor **composition** over inheritance. You also know that both `class` and `extends` keywords are only syntactic sugar and there is [a better option](https://medium.com/javascript-scene/the-two-pillars-of-javascript-ee6f3281e7f3#d03c) of instantiating new objects. This option is called **factory function** which allows you to **compose** any objects you need into new prototype.
 
 **Bad:**
 ```javascript
-class Employee {
-  constructor(name, email) {
+class Airplane {
+  constructor(initialState) {
+    const { name } = initialState;
+
     this.name = name;
-    this.email = email;
   }
 
-  // ...
+  getName() {
+    return `${this.name}`;
+  }
 }
 
-// Bad because Employees "have" tax data. EmployeeTaxData is not a type of Employee
-class EmployeeTaxData extends Employee {
-  constructor(ssn, salary) {
-    super();
-    this.ssn = ssn;
-    this.salary = salary;
+const defaultAirbusA380State = {
+  name: 'Airbus A380'
+};
+class AirbusA380 extends Airplane {
+  constructor(initialState) {
+    const state = Object.assign(
+      {},
+      defaultAirbusA380State,
+      initialState
+    );
+    const { airline } = state;
+
+    super(state);
+
+    this.airline = airline;
   }
 
-  // ...
+  getName() {
+    return  `${super.getName()} of ${this.airline}`;
+  }
 }
-```
 
-**Good**:
-```javascript
-class EmployeeTaxData {
-  constructor(ssn, salary) {
-    this.ssn = ssn;
-    this.salary = salary;
-  }
+const defaultCessnaState = {
+  name: 'Cessna'
+};
+class Cessna extends Airplane {
+  constructor(initialState) {
+    const state = Object.assign(
+      {},
+      defaultCessnaState,
+      initialState
+    );
 
-  // ...
+    super(state);
+  }
 }
 
-class Employee {
-  constructor(name, email) {
-    this.name = name;
-    this.email = email;
+// create instances of airplanes
+const emiratesAirbus = new AirbusA380({
+  airline: 'Emirates'
+});
+
+const privateCessnaJet = new Cessna();
+
+console.log(emiratesAirbus.getName()); // prints "Airbus A380 of Emirates"
+console.log(privateCessnaJet.getName()); // prints "Cessna"
+```
+
+**Good:**
+```javascript
+// define base objects
+const Airplane = {
+  getName() {
+    return `${this.name}`;
```

Comment:
I left it like this for the consistency with `AirbusA380.getName` method, but sure it's better to have shorter form here as the present one brings nothing and adds a few more characters.